src/plugins/intel_gpu/src/graph/impls/ocl_v2/moe/moe_3gemm_gen_micro.cpp

```diff
@@ -0,0 +1,446 @@
+// Copyright (C) 2025 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#ifdef ENABLE_ONEDNN_FOR_GPU
+// clang-format off
+// Put this file at first to avoid incorrect header files includes order.
+// For example, intel_gpu/runtime/utils.hpp will causes compiling error in hash<dnnl::impl::primitive_hashing::key_t>
+#include "moe_3gemm_gen_micro.hpp"
+
+#include "intel_gpu/graph/kernel_impl_params.hpp"
+#include "intel_gpu/primitives/moe_3gemm_fused_compressed.hpp"
+#include "ocl_v2/utils/jitter.hpp"
+#include "../utils/kernel_generator.hpp"
+
+// clang-format on
+namespace ov::intel_gpu::ocl {
+
+static size_t get_subgroup_size(gpu_arch arch) {
+    switch (arch) {
+    case gpu_arch::gen9:
+    case gpu_arch::gen11:
+    case gpu_arch::xe_lp:
+    case gpu_arch::xe_hp:
+    case gpu_arch::xe_hpg:
+        return 8;
+    case gpu_arch::xe_hpc:
+    case gpu_arch::xe2:
+    case gpu_arch::xe3:
+        return 16;
+    default:
+        // fallback to safe value
+        return 8;
+    }
+}
+
+JitConstants MoE3GemmMicroGenerator::get_jit_constants(const kernel_impl_params& params, const micro::Package& moe_gemm, const moe_3gemm_config& cfg) const {
+    const auto& device_info = params.get_device_info();
+    const size_t subgroup_size = get_subgroup_size(device_info.arch);
+    const auto& weight_layout = params.input_layouts[m_wei_idx];
+    const auto& scale_layout = params.input_layouts[m_scale_idx];
+    const auto& zp_layout = params.input_layouts[m_zp_idx];
+
+    // Internal generator of JIT constants, require intermediate buffers and part of primitive's inputs.
+    JitConstants jit;
```

Comment:
```suggestion
    auto jit = make_base_jit_constants(params);
```
I think you can remove several jit definitions by using this.

Reply:
Considered it requires intermediate memories and parameter's dynamic shape will introduce `OPTIONAL_SHAPE_INFO_ARG` which is not needed and causes kernel compiler error, so I think `make_base_jit_constants` cannot meet such usage.